### ISSUE TITLE
Add CODEOWNERS file to MDTF-diagnostics repo

### ISSUE THREAD

**Issue by wrongkindofdoctor** [NOAA-GFDL/MDTF-diagnostics#53]
Adding a CODEOWNERS file to the .github folder will automate assigning developers to review PRs, and require at least one codeowner to approve a PR before it is merged into the development branch. Owners may be defined for the repository as a whole, and/or for different directories. The [FMS CODEOWNERS file](https://github.com/NOAA-GFDL/FMS/blob/master/.github/CODEOWNERS) may be used as a template

**Comment by jkrasting** [comment]
Great idea, @wrongkindofdoctor - this will help the code review process going forward

**Comment by tsjackson-noaa** [comment]
Thanks @wrongkindofdoctor - this will be really useful in connection with the new open-source security policy. If I understand the [docs](https://docs.github.com/en/free-pro-team@latest/github/creating-cloning-and-archiving-repositories/about-code-owners) correctly, @jkrasting will need to do this (owner or admin permissions needed). 

It would've been nice if we could use this mechanism to ping the point of contact for each POD's development in, but it looks like everyone listed needs write access.

**Comment by wrongkindofdoctor** [comment]
Yes @tsjackson-noaa,  you are correct about the access permissions for external collaborators.  To ensure that the repo admins are the only ones merging into the development branch, the repo owner could create a rule (see [here](https://docs.github.com/en/free-pro-team@latest/github/administering-a-repository/enabling-required-reviews-for-pull-requests)) to require reviews from 2 (or more) codeowners before a branch is merged: the first owner would be the POC, and the other owners would be you, @jkrasing, and/or me.

**Comment by jkrasting** [comment]
I enabled code reviews for the `develop` branch where we instructed the POD developers to submit their PRs.  I also added the CODEOWNERS file and added the three of us.

I made a dummy PR editing the README.md file -- #59.  @wrongkindofdoctor, @tsjackson-noaa - did either of you get pinged for a code review?

**Comment by wrongkindofdoctor** [comment]
@jkrasting Nope. From the [documentation](https://docs.github.com/en/enterprise-server@2.19/github/creating-cloning-and-archiving-repositories/about-code-owners):

> Each CODEOWNERS file assigns the code owners for a single branch in the repository

You need a CODEOWNERS file in the develop branch for this PR to ping me and Tom; the CODEOWNERS file is only in the main branch.

**Comment by tsjackson-noaa** [comment]
@jkrasting : yes, received the ping.

I believe the docs still say to submit PRs to ``develop``, but I've been led to make individual permanent dev branches for each pod under ``pod/<POD's name>``. This permits testing and updates to each POD to be done in parallel and makes sure the history only contains commits specific to each POD. It's no issue in practice, as the target branch of a PR may be edited.

**Comment by jkrasting** [comment]
Ah I see.  We can write rules - should we apply the review instead to `pod/*` branches?

**Comment by jkrasting** [comment]
@wrongkindofdoctor - I added CODEOWNERS to the develop branch.  I closed @tsjackson-noaa's PR (#60) and reopened.  Did you get a review request?  It now explicitly mentions both of us as needing to review the code.

**Comment by wrongkindofdoctor** [comment]
@jkrasting No. Did you check the *require review from codeowners* box in the branch protection rules settings?
